lib/galaxy/webapps/galaxy/controllers/admin.py

```diff
@@ -1600,14 +1600,25 @@ def manage_roles_and_groups_for_user(self, trans, payload=None, **kwd):
             return {'message' : 'User \'%s\' has been updated with %d associated roles and %d associated groups (private roles are not displayed).' % (user.email, len(in_roles) - 1, len(in_groups))}
 
     @web.expose
+    @web.json
     @web.require_admin
-    def jobs(self, trans, stop=[], stop_msg=None, cutoff=180, job_lock=None, ajl_submit=None, **kwd):
+    def jobs_control(self, trans, job_lock=None, **kwd):
+        if job_lock is not None:
+            job_lock = True if job_lock == 'true' else False
+            galaxy.queue_worker.send_control_task(trans.app, 'admin_job_lock', kwargs={'job_lock': job_lock})
+        job_lock = trans.app.job_manager.job_lock
+        return {'job_lock': job_lock}
+
+    @web.expose
+    @web.json
+    @web.require_admin
+    def jobs_list(self, trans, stop=[], stop_msg=None, cutoff=180, **kwd):
         deleted = []
-        msg = None
-        status = None
+        message = kwd.get('message', '')
+        status = kwd.get('status', 'info')
         job_ids = util.listify(stop)
         if job_ids and stop_msg in [None, '']:
-            msg = 'Please enter an error message to display to the user describing why the job was terminated'
+            message = 'Please enter an error message to display to the user describing why the job was terminated'
             status = 'error'
```

Comment:
Alternatively you could use: `return self.message_exception(trans, message)` and then catch the error message on the client in the exception of the `axios` request.